src/common.jl

```diff
@@ -166,25 +167,38 @@ bordermap[:dashed] = border_dashed
 bordermap[:dotted] = border_dotted
 bordermap[:ascii]  = border_ascii
 
+const UserColorType = Union{Integer,Symbol,NTuple{3,Integer}}
+const ColorType = Union{Nothing,UInt8}
+
 const color_cycle = [:green, :blue, :red, :magenta, :yellow, :cyan]
-const color_encode = Dict{Symbol,UInt8}()
-const color_decode = Dict{UInt8,Symbol}()
-color_encode[:normal]  = 0b000
-color_encode[:blue]    = 0b001
-color_encode[:red]     = 0b010
-color_encode[:magenta] = 0b011
-color_encode[:green]   = 0b100
-color_encode[:cyan]    = 0b101
-color_encode[:yellow]  = 0b110
-for k in keys(color_encode)
-    v = color_encode[k]
-    color_decode[v] = k
+
+function print_color(color::Union{Nothing,Integer,Symbol}, io::IO, args...)
+    col = if color === nothing
+        :normal
+    elseif color isa Symbol
+        color
+    else
+        Int(color)
+    end
+    printstyled(io, string(args...); color = col)
 end
-color_encode[:white] = 0b111
-color_decode[0b111]  = :white
 
-function print_color(color::UInt8, io::IO, args...)
-    col = color in keys(color_decode) ? color_decode[color] : Int(color)
-    str = string(args...)
-    printstyled(io, str; color = col)
+function crayon_256_color(color::UserColorType)::ColorType
```

Comment:
This seems a bit hacky, should I PR to `Crayons.jl` instead ?

Reply:
Proposing a stable API in Crayons sounds okay to me but I know very little about this. I don't know if it's useful, there are some discussions on https://github.com/JuliaGraphics/Colors.jl/issues/473

Reply:
Thanks for the link. What we need in `UnicodePlots` is something generic working on all terminals.

@KristofferC, do you have any advice for this ?